examples/dmd/local_tw_csv.cpp

```diff
@@ -0,0 +1,640 @@
+            double tval = tvec[idx_snap];
+            string data_filename = string(data_dir) + "/" + par_dir + "/" + snap + "/" + variable + ".csv"; // path to VAR_NAME.csv
+            csv_db.getDoubleArray(data_filename, sample, nelements, idx_state);
+            if (myid == 0)
+            {
+                cout << "State " << data_filename << " read." << endl;
+            }
+
+            if (idx_snap == 0)
+            {
+                dmd_preprocess_timer.Start();
+                CAROM::Vector* init_cond = nullptr;
+                for (int window = 0; window < numWindows; ++window)
+                {
+                    if (myid == 0)
+                    {
+                        cout << "Projecting initial condition at t = " << indicator_val[window] << " for DMD model #" << window << endl;
+                    }
+                    if (window == 0)
+                    {
+                        init_cond = new CAROM::Vector(dim, true);
+                        for (int i = 0; i < dim; ++i)
+                        {
+                            init_cond->item(i) = sample[i];
+                        }
+                    }
+                    else
+                    {
+                        init_cond = dmd[window-1]->predict(indicator_val[window]);
+                    }
+                    dmd[window]->projectInitialCondition(init_cond);
+                    delete init_cond;
+                }
+                dmd_preprocess_timer.Stop();
+            }
+
+            if (t_final > 0.0) // Actual prediction without true solution for comparison
+            {
+                num_tests += 1;
+                while (curr_window+1 < numWindows && t_final > indicator_val[curr_window+1])
+                {
+                    curr_window += 1;
+                }
+                if (myid == 0)
+                {
+                    cout << "Predicting DMD solution at t = " << t_final << " using DMD model #" << curr_window << endl;
+                }
+                dmd_prediction_timer.Start();
+                CAROM::Vector* result = dmd[curr_window]->predict(t_final);
+                dmd_prediction_timer.Stop();
+                if (myid == 0)
+                {
+                    csv_db.putDoubleArray(outputPath + "/" + par_dir + "_final_time_prediction.csv", result->getData(), dim);
+                }
+                idx_snap = snap_bound[1]+1; // escape for-loop over idx_snap
+                delete result;
+            }
+            else // Verify DMD prediction results against dataset
+            {
+                while (curr_window+1 < numWindows && tval > indicator_val[curr_window+1])
+                {
+                    curr_window += 1;
+                }
+                if (myid == 0)
+                {
+                    cout << "Predicting DMD solution #" << idx_snap << " at t = " << tval << " using DMD model #" << curr_window << endl;
```

Comment:
I don't think libROM has any policy about line length, but we generally try to avoid very long lines (e.g. over 100 characters). This file has several long lines that could be broken up.

Reply:
It looks like astyle has an option to enforce line length. What do you think about using this option?

Reply:
MFEM enforces 80-character lines, and I think it would be good to have a limit, if only to simplify reviewing (no more wondering if a line is too long, just make it automatic).

Reply:
I've added a 80-character limit to stylize, and it causes changes on few hundred lines. Is it okay to push it here when you are reviewing, or I should wait until you finish the review @dylan-copeland ?

Reply:
For simplicity, I would prefer a separate PR just for the max line length. This PR can have long lines.

Reply:
Sounds good to me.